apps/class-solid/src/components/Experiment.tsx

```diff
@@ -89,40 +109,74 @@ function RunningIndicator() {
   );
 }
 
-function DownloadExperiment(props: { experiment: Experiment }) {
+function DownloadExperimentConfiguration(props: { experiment: Experiment }) {
   const downloadUrl = createMemo(() => {
-    // Drop id and running
-    const data = {
-      name: props.experiment.name,
-      description: props.experiment.description,
-      config: props.experiment.config,
-      output: props.experiment.output,
-    };
-    return URL.createObjectURL(
-      new Blob([JSON.stringify(data, undefined, 2)], {
-        type: "application/json",
-      }),
-    );
+    return URL.createObjectURL(toConfigBlob(props.experiment));
   });
 
   onCleanup(() => {
     URL.revokeObjectURL(downloadUrl());
   });
 
   const filename = `class-${props.experiment.id}.json`;
-
   return (
-    <a
-      class={buttonVariants({ variant: "outline" })}
-      href={downloadUrl()}
-      download={filename}
-      type="application/json"
-    >
-      <MdiDownload />
+    <a href={downloadUrl()} download={filename} type="application/json">
+      Configuration
     </a>
   );
 }
 
+function DownloadExperimentArchive(props: { experiment: Experiment }) {
+  const [downloadUrl] = createResource(props.experiment, async (experiment) => {
+    const archive = await createArchive(experiment);
+    return URL.createObjectURL(archive);
+  });
+
+  onCleanup(() => {
+    if (downloadUrl.latest) {
+      URL.revokeObjectURL(downloadUrl.latest);
+    }
+  });
+
+  const filename = `class-${props.experiment.id}.zip`;
```

Comment:
I would prefer to use the experiment name, but again, can lead to conflicts.

Reply:
can do